src/pipeline/src/etl/processor/regex.rs

```diff
@@ -203,6 +237,48 @@ impl Processor for RegexProcessor {
             )),
         }
     }
+
+    fn exec_mut(&self, val: &mut Vec<Value>) -> Result<(), String> {
+        for field in self.fields.iter() {
```

Comment:
The code block appears to be similar across different processors. Perhaps we can utilize a macro to eliminate the duplications.

Reply:
My suggestion is to avoid macro when it's possible. Macro can make the code difficult to understand.

Reply:
However, these function implementations appear identical. If you want to change one place, you have to change almost everywhere. Maybe not macro but some way to reduce the duplicated code.

Reply:
Next phase here will be another design to ensure code readability and reusability.